scripts/npm-safe-publish.sh

```diff
@@ -67,35 +92,30 @@ if [[ "$NPM_SHOW" != "$NPM_LS" ]]; then
       echo "Working directory is not clean. Aborting."
       exit 1
     fi
+  fi
 
-    # check if current commit is tagged with the requested version
-    if [[ -z "$(git tag --points-at HEAD "$PACKAGE@$NPM_LS")" ]]; then
-      echo "Current commit is not tagged with the requested version. Aborting."
-      exit 1
-    fi
-
-    # check if otp is set
-    if [[ -z "$OTP" ]]; then
-      echo "Must specify the NPM one-time password using the --otp option."
-      exit 1
-    fi
+  # check if current commit is tagged with the requested version
+  if [[ -z "$(git tag --points-at HEAD "$PACKAGE@$NPM_LS")" ]]; then
+    echo "Current commit is not tagged with the requested version. Aborting."
+    exit 1
+  fi
 
+  if [[ -z "$DRY_RUN" ]]; then
     echo "Publishing $PACKAGE@$NPM_LS to NPM."
-    npm publish --otp=$OTP
-
-    if [[ "$?" != 0 ]]; then
-      echo "Publish failed. Aborting."
-      exit 1
-    fi
-
-    curl -X POST \
-      -d api_key=$DEVCYCLE_PROD_SLEUTH_API_TOKEN \
-      -d environment=production \
-      -d sha=$SHA https://app.sleuth.io/api/1/deployments/taplytics/js-sdks-2/register_deploy
+    npm_authenticated publish
   else
-    echo "NPM Publish Local"
-    npm publish --local
+    echo "[DRY RUN] Not publishing $PACKAGE@$NPM_LS to NPM."
   fi
+
+  if [[ "$?" != 0 ]]; then
+    echo "Publish failed. Aborting."
+    exit 1
+  fi
+
+  curl -X POST \
```

Comment:
We can remove all the sleuth stuff since this is now able to be tracked as a github action in Sleuth directly.

Reply:
Leaving this in for now because I don't understand the sleuth configuration, and it's apparently working the way it is.

Reply:
Frankly neither do I, I just get Nik to do it